docs/architecture/microservices/multi-container-microservice-net-applications/database-server-container.md

```diff
@@ -47,22 +47,107 @@ Having SQL Server running as a container is not just useful for a demo where you
 
 ### Seeding with test data on Web application startup
 
-To add data to the database when the application starts up, you can add code like the following to the Configure method in the Startup class of the Web API project:
+To add data to the database when the application starts up, you can add code like the following to the `Main` method in the `Program` class of the Web API project:
 
 ```csharp
-public class Startup
+public static int Main(string[] args)
+{
+    var configuration = GetConfiguration();
+
+    Log.Logger = CreateSerilogLogger(configuration);
+
+    try
+    {
+        Log.Information("Configuring web host ({ApplicationContext})...", AppName);
+        var host = CreateHostBuilder(configuration, args);
+
+        Log.Information("Applying migrations ({ApplicationContext})...", AppName);
+        host.MigrateDbContext<CatalogContext>((context, services) =>
+        {
+            var env = services.GetService<IWebHostEnvironment>();
+            var settings = services.GetService<IOptions<CatalogSettings>>();
+            var logger = services.GetService<ILogger<CatalogContextSeed>>();
+
+            new CatalogContextSeed()
+                .SeedAsync(context, env, settings, logger)
+                .Wait();
+        })
+        .MigrateDbContext<IntegrationEventLogContext>((_, __) => { });
+
+        Log.Information("Starting web host ({ApplicationContext})...", AppName);
+        host.Run();
+
+        return 0;
+    }
+    catch (Exception ex)
+    {
+        Log.Fatal(ex, "Program terminated unexpectedly ({ApplicationContext})!", AppName);
+        return 1;
+    }
+    finally
+    {
+        Log.CloseAndFlush();
+    }
+}
+```
+
+There's an important caveat when applying migrations and seeding a database during container startup. Since the database server might not be available for whatever reason, so you have to handle retries while waiting for the server to be available. This retry logic is handle by the `MigrateDbContext()` extension method, as shown in the following code:
```

Comment:
```suggestion
There's an important caveat when applying migrations and seeding a database during container startup. Since the database server might not be available for whatever reason, you must handle retries while waiting for the server to be available. This retry logic is handled by the `MigrateDbContext()` extension method, as shown in the following code:
```